mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/HistoryEvolution.java

```diff
@@ -0,0 +1,106 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.quota;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.james.mailbox.quota.model.QuotaThreshold;
+
+public class HistoryEvolution {
+
+    public static HistoryEvolution noChanges(QuotaThreshold currentThreshold) {
+        return new HistoryEvolution(ThresholdHistoryChange.NoChange,
+            Optional.empty(),
+            currentThreshold);
+    }
+
+    public static HistoryEvolution lowerThresholdReached(QuotaThreshold currentThreshold) {
+        return new HistoryEvolution(ThresholdHistoryChange.LowerThresholdReached,
+            Optional.empty(),
```

Comment:
We do care. Bur not the way you intend it.

What we do care is to know if that threshold (or higher) had been reached within a certain period. If so, the user had already been notified. More likely we are getting around the threshold. The goal here is to avoid spamming for a given level.

Reply:
We didn't understood each others : I meant, for lowerThresholdReached, you should not have empty as recentness, right ?